### ISSUE TITLE
Evaluation is not consistent between  model.eval_model and model.predict

### ISSUE THREAD

**Issue by ThiagoSousa** [ThilinaRajapakse/simpletransformers#1176]
I have trained a bert-base-cased ClassificationModel for binary classification and evaluating it on some test data regarding precision, recall and F1 scores. The problem is that I found out a difference in the evaluation between the results produced by model.predict and model.eval_model for the ClassificationModel. 

First, my code: 

```
#Training
model = ClassificationModel('bert', 'bert-base-cased', args=args)
model.train_model(train_df)

#Evaluation
result, model_outputs, wrong_predictions = model.eval_model(test_df)
predictions = np.argmax(model_outputs,axis=1)

# 1- precision and recall on the sklearn.metrics.classification_report
report = classification_report(test_df["label"].tolist(), predictions, target_names={"Wrong":0,"Correct":1})
print(report)

# 2- precision and recall calculated manually from the output of the eval_model
precision = result["tp"]/(result["tp"]+result["fp"])
recall = result["tp"]/(result["tp"]/result["fn"])
print(precision)
print(recall)
print(result)

# 3- directly predicting from the model and evaluating separately
test = test_df.to_numpy()
test_input = test[:,[0,1]]
test_class = test[:,[2]]
predictions, raw_outputs = model.predict(test_input.tolist())
report = classification_report(test_class, predictions, target_names={"Wrong":0,"Correct":1})
```

**Expected behavior**
I was expecting all three to give me the same results, but in fact I get:

1- the first evaluation from the predictions of eval_model gives me in model_outputs:

```
precision    recall  f1-score   support

       Wrong       0.85      0.86      0.86      1753
     Correct       0.14      0.14      0.14       296

    accuracy                           0.75      2049
   macro avg       0.50      0.50      0.50      2049
weighted avg       0.75      0.75      0.75      2049
```

2- the precision and recall that I calculate from the result of eval_model
`PRECISION 0.7638888888888888
RECALL 0.7638888888888888
`

3- The precision and recall from predicting the docs directly

```
precision    recall  f1-score   support

       Wrong       0.99      0.98      0.98      1753
     Correct       0.87      0.94      0.91       296

    accuracy                           0.97      2049
   macro avg       0.93      0.96      0.94      2049
weighted avg       0.97      0.97      0.97      2049
```

So why is there a different difference in the evaluation frmo eval_model and predict? Either I'm doing something very wrong or there is a problem with the library. I tend to think that the sklearn is correct and that's is not the problem. And what I pass to eval_model and predict is also correct, unless eval_model somehow shuffles this data.
